### PR TITLE
image hash work around

### PR DESCRIPTION
**Pull request by marqh**
temporary work around for unknown and occasional non-repeatable check_graphic test failure

if it fails, try again, mostly it works second time round

masks #2195 does not fix it

### CODE REVIEW

**Comment by marqh** [comment]
@bjlittle it's not pretty, but it may be better than nothing for now

File: lib/iris/tests/__init__.py [hunk 1]
```diff
@@ -702,13 +700,17 @@ def check_graphic(self, tol=None):
                     if err.errno != 17:
                         raise
 
-            figure.savefig(result_fname)
+            def save_figure_hash():
+                figure = plt.gcf()
+                figure.savefig(result_fname)
```
Comment (bjlittle) [inline lib/iris/tests/__init__.py]: @marqh `plt.gcf().savefig(result_fname)` works for me ... but I don't see the point in making such a cosmetic change here in this PR.

I'm going to have to rebase #2192 with this anyways, so I can tweak away in my PR :wink:

**Comment by bjlittle** [comment]
@marqh Agreed that this is a short-term fix that's not going to close #2195 :+1:

File: lib/iris/tests/__init__.py (hunk 1 above)
Comment (marqh) [inline lib/iris/tests/__init__.py]: I don't think this is a cosmetic change

previously the `figure = plt.gcf()` and `figure.savefig(result_fname)` were in the code, a little way apart.

i moved them together, without changing either line

minor point tho

**Comment by bjlittle** [comment]
@marqh Unfortunately, I don't see this `fix` making much difference ... we still have this issue, or at least I'm still encountering it with this PR included :cry:

**Pull request merged**